aws_lambda_powertools/shared/cookies.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime
+from enum import Enum
+from io import StringIO
+from typing import List, Optional
+
+
+class SameSite(Enum):
+    DEFAULT_MODE = ""
+    LAX_MODE = "Lax"
+    STRICT_MODE = "Strict"
+    NONE_MODE = "None"
+
+
+def _getdate(timestamp: datetime) -> str:
+    return timestamp.strftime("%a, %d %b %Y %H:%M:%S GMT")
+
+
+class Cookie:
+    def __init__(
+        self,
+        name: str,
+        value: str,
+        path: Optional[str] = None,
+        domain: Optional[str] = None,
+        expires: Optional[datetime] = None,
+        max_age: Optional[int] = None,
+        secure: Optional[bool] = None,
+        http_only: Optional[bool] = None,
+        same_site: Optional[SameSite] = None,
+        custom_attributes: Optional[List[str]] = None,
+    ):
+        self.name = name
+        self.value = value
+        self.path = path
+        self.domain = domain
+        self.expires = expires
+        self.max_age = max_age
+        self.secure = secure
+        self.http_only = http_only
+        self.same_site = same_site
+        self.custom_attributes = custom_attributes
+
+    def __str__(self) -> str:
```

Comment:
For the length of string this is fine, but careful with `str` as it's one the slowest part in Python as it's actually a Class, and every call to `str()` creates an object + permutations.

I ran a quick benchmark and this part of the code is ~6ns so it's more than fine.